message_processor.go

```diff
@@ -27,11 +31,40 @@ func (mg messageProcessor) process(lf *logPair) {
 	consumeLog(lf)
 }
 
+type tmpBuffer struct {
+	tBuf                      bytes.Buffer
+	bufferTimer               time.Time
+	bufferReset               bool
+	bufferHoldDurationExpired bool
+}
+
+func appendToTempBuffer(p *tmpBuffer, b *logdriver.LogEntry) {
+	// Add msg to temp buffer and disable buffer reset flag
+	ps, err := p.tBuf.Write(b.Line)
+	p.bufferReset = false
+	if err != nil {
+		logrus.WithError(err).WithField("Appending to Temp Buffer with size:", ps).Error(
+			"Error appending to buffer")
+	}
+}
+
+func resetTempBufferValues(p *tmpBuffer) {
```

Comment:
it appears to me this should be a member function

```
func (b *tmpBuffer) Reset() {
    ...
}
```

Reply:
Done